android/pytorch_android/src/main/java/org/pytorch/Tensor.java

```diff
@@ -0,0 +1,286 @@
+package org.pytorch;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.util.Arrays;
+import java.util.Locale;
+
+public abstract class Tensor {
+  private static final int TYPE_CODE_BYTE = 1;
+  private static final int TYPE_CODE_INT32 = 2;
+  private static final int TYPE_CODE_FLOAT32 = 3;
+
+  private static final String ERROR_MSG_DATA_BUFFER_NOT_NULL = "Data buffer must be not null";
+  private static final String ERROR_MSG_DATA_ARRAY_NOT_NULL = "Data array must be not null";
+  private static final String ERROR_MSG_DIMS_NOT_NULL = "Dims must be not null";
+  private static final String ERROR_MSG_DIMS_NOT_EMPTY = "Dims must be not empty";
+  private static final String ERROR_MSG_INDEX_NOT_NULL = "Index must be not null";
+  private static final String ERROR_MSG_DIMS_NON_NEGATIVE = "Dims must be non negative";
+  private static final String ERROR_MSG_DATA_BUFFER_MUST_HAVE_NATIVE_BYTE_ORDER =
+      "Data buffer must have native byte order (java.nio.ByteOrder#nativeOrder)";
+  private static final String ERROR_MSG_DATA_BUFFER_MUST_BE_DIRECT =
+      "Data buffer must be direct (java.nio.ByteBuffer#allocateDirect)";
+
+  public final long[] dims;
+
+  private static final int FLOAT_SIZE_BYTES = 4;
+  private static final int INT_SIZE_BYTES = 4;
+
+  public static FloatBuffer allocateFloatBuffer(int numElements) {
+    return ByteBuffer.allocateDirect(numElements * FLOAT_SIZE_BYTES).order(ByteOrder.nativeOrder()).asFloatBuffer();
+  }
+
+  public static IntBuffer allocateIntBuffer(int numElements) {
+    return ByteBuffer.allocateDirect(numElements * INT_SIZE_BYTES).order(ByteOrder.nativeOrder()).asIntBuffer();
+  }
+
+  public static ByteBuffer allocateByteBuffer(int numElements) {
+    return ByteBuffer.allocateDirect(numElements).order(ByteOrder.nativeOrder());
+  }
+
+  public static Tensor newFloatTensor(long[] dims, float[] data) {
+    checkArgument(data != null, ERROR_MSG_DATA_ARRAY_NOT_NULL);
+    checkArgument(dims != null, ERROR_MSG_DIMS_NOT_NULL);
+    checkDims(dims);
+    checkDimsAndDataCapacityConsistency(data.length, dims);
+    final int bufferCapacity = (int) numElements(dims);
+    final FloatBuffer floatBuffer = allocateFloatBuffer(bufferCapacity);
+    floatBuffer.put(data);
+    return new Tensor_float32(floatBuffer, dims);
+  }
+
+  public static Tensor newIntTensor(long[] dims, int[] data) {
+    checkArgument(data != null, ERROR_MSG_DATA_ARRAY_NOT_NULL);
+    checkArgument(dims != null, ERROR_MSG_DIMS_NOT_NULL);
+    checkDims(dims);
+    checkDimsAndDataCapacityConsistency(data.length, dims);
+    final int bufferCapacity = (int) numElements(dims);
+    final IntBuffer intBuffer = allocateIntBuffer(bufferCapacity);
+    intBuffer.put(data);
+    return new Tensor_int32(intBuffer, dims);
+  }
+
+  public static Tensor newByteTensor(long[] dims, byte[] data) {
```

Comment:
can we follow the same names at C++ API whenever possible? There's at::from_blob?